artifactory/commands/transferinstall/installplugin.go

```diff
@@ -0,0 +1,356 @@
+		pluginName:         pluginName,
+		installVersion:     nil, // latest
+		localSrcDir:        "",
+		localJfrogHomePath: "",
+	}
+}
+
+func (ipc *InstallPluginCommand) ServerDetails() (*config.ServerDetails, error) {
+	return ipc.targetServer, nil
+}
+
+// Set the local file system directory path that the plugin files will be copied from
+func (ipc *InstallPluginCommand) SetLocalPluginFiles(localDir string) *InstallPluginCommand {
+	ipc.localSrcDir = localDir
+	return ipc
+}
+
+// Set the plugin version we want to download
+func (ipc *InstallPluginCommand) SetInstallVersion(installVersion *version.Version) *InstallPluginCommand {
+	ipc.installVersion = installVersion
+	return ipc
+}
+
+// Set the base URL that the plugin files avaliable
+func (ipc *InstallPluginCommand) SetBaseDownloadUrl(baseUrl string) *InstallPluginCommand {
+	ipc.baseDownloadUrl = baseUrl
+	return ipc
+}
+
+// Set the Jfrog home directory path override to search in it the plugin directory as destination
+func (ipc *InstallPluginCommand) SetJFrogHomePath(path string) *InstallPluginCommand {
+	ipc.localJfrogHomePath = path
+	return ipc
+}
+
+// PluginInstallManager holds all the plugin files and optional plugin directory destinations.
+// we construct the destination path by searching for the existing plugin directory destination from the options
+// and joining the local path of the plugin file with the plugin directory
+type PluginInstallManager struct {
+	// All the plugin files, represented as FileItem with path starting from inside the plugin directory
+	// i.e. for file plugins/lib/file we represent them as FileItem{"lib","file"}
+	files PluginFiles
+	// All the optional destinations of the plugin directory starting from the JFrog home directory represented as FileItem
+	destinations []FileItem
+}
+
+// Create new file transfer manager for artifactory plugins
+func NewArtifactoryPluginInstallManager(bundle PluginFiles) *PluginInstallManager {
+	manager := &PluginInstallManager{
+		files:        bundle,
+		destinations: []FileItem{},
+	}
+	// Add all the optional destinations for the plugin dir
+	manager.addDestination(originalDirPath)
+	manager.addDestination(v7DirPath)
+	return manager
+}
+
+// Add optional plugin directory location as destination
+func (ftm *PluginInstallManager) addDestination(directory FileItem) {
+	ftm.destinations = append(ftm.destinations, directory)
+}
+
+// Search all the local target directories that the plugin directory can exist in base on a given root JFrog artifactory home directory
+// we are searching by Joining rootDir/optionalDestination to find the right structure of the JFrog home directory
+// the first option that matched and the directory exists is returned as target
+func (ftm *PluginInstallManager) trySearchDestinationMatchFrom(rootDir string) (exists bool, target FileItem, err error) {
+	if exists, err = fileutils.IsDirExists(rootDir, false); err != nil || !exists {
+		return
+	}
+	exists = false
```

Comment:
its not, because I use the same variable before for checking if the root dir exists, if it does not it will return so it will always be true when getting to this line